### DISCUSSION TITLE
can not subscribe ros message with foxglove bridge

### DISCUSSION THREAD

**Discussion by kenjiWuJian** [foxglove/community#998] [Bugs]
Environment: 
ROS2 galactic running in docker
foxglove sttudio and foxglove bridge
Issue:
foxglove bridge report error as below:

[foxglove_bridge-1] [INFO] [1715136565.775747788] [foxglove_bridge]: Adding subscriber #0 to topic "/TrajectoryViz" (visualization_msgs/msg/MarkerArray) on channel 12
[foxglove_bridge-1] [ERROR] [1715136565.776080755] [foxglove_bridge]: Failed to subscribe to topic "/TrajectoryViz" (visualization_msgs/msg/MarkerArray): Failed to initialize event: provided event_type is not supported by rmw_fastrtps_cpp, at /tmp/binarydeb/ros-galactic-rmw-fastrtps-shared-cpp-5.0.2/src/rmw_event.cpp:59

I understand maybe it is not the foxglove issue, however, I don't know how to solve this and it is hard to find a solution online. Could you help me with your expertise? Thank you.

**Comment by defunctzombie** [comment]
This sounds like a ROS2 issue with something about the fastrtps or middleware. Hard to say more without better steps to reproduce.

Unfortunately, even with good reproduction steps, because ROS2 Galactic has reached EOL (https://docs.ros.org/en/iron/Releases.html) in Dec 2022 finding good support for fixes and changes might be hard.

**Reply by kenjiWuJian** [reply]
Thank you for your support. ROS bridge can work and I will use it instead.
